fix(mobile): Generate 1 splash screen on Android

martyfuhry opened the pull request:
- Adds `flutter_native_splash` dependency and build script 
- Required adding `image:` to the `dependency_overrides` since `flutter_launcher_icons` is out of date
- Adds `create_splash` command to `makefile` to run the `flutter_native_splash` create script
- Adds `assets/immich-logo-no-outline-android12.png` which is 1152x1152 pixels with the app icon centered within a 640 pixel diameter circle to match the [Android 12 splash screens](https://developer.android.com/develop/ui/views/launch/splash-screen) requirements
- And changes initial splash screen to no longer contain the "Immich" text.

The initial splash screen no longer contains the text of the app name since that would require adding the text to the splash screen image. If that's desired, I can work on that.

Closes #982

alextran1502 commented on mobile/ios/Runner/Info.plist:
```diff
@@ -1,105 +1,97 @@
+			<string>sk</string>
+			<string>zh</string>
+		</array>
+		<key>UIStatusBarHidden</key>
```
Is this auto generated by some process in this PR?

martyfuhry replied:
Odd, it might be. Let me take a look.

martyfuhry replied:
Yup, I just reran the script `flutter pub run flutter_native_splash:create` (from the `makefile`) and this line shows up in the output:

`[iOS] Updating ios/Runner/Info.plist for status bar hidden/visible`

I suppose that answers our question. It sure generates a terrible diff. I think it's just an indentation, but I didn't read it very carefully. The main relevant line it seems to be adding is here at the end:

```
<key>UIStatusBarHidden</key>
<false/>
```

alextran1502 commented:
Can you help me fix the `Info.plist` conflict?

martyfuhry committed:
changed splash screen and added flutter_native_splash and android 12 splash screens

martyfuhry commented:
I rebased on `main`, that should do it. The only real (non-generated) changes to this PR are:

1. Added `flutter_native_splash` as a `dev_dependency`
2. Added `create_splash` to the `makefile`
3. Changed the `SplashScreenPage` to no longer show the "Immich" text (which may be a bigger change than you like)

Everything else was generated by `make create_splash`.

If you'd like to remain using the "Immich" text, you'll need to add that to the initial splash screen image to avoid the "double splash screen" effect from the inherent issue. You want your splash asset to exactly match the first frame that Flutter is going to draw in your app, which happens to be the `SplashScreenPage` in order to appear seamless. 

Someone should also test iOS to make sure the splash screens / assets look OK to them.

martyfuhry commented:
https://user-images.githubusercontent.com/100457/215288504-f7b6c88b-9c6f-4894-8a97-01f47cc099fb.mp4

Here's how it looks on my Android 12 device. Notice how the splash screen exactly matches the first frame from Immich. You could also animate the Splash to Login transition with a Hero animation pretty easily.
